setup/src/Magento/Setup/Model/ConfigGenerator.php

```diff
@@ -70,15 +81,14 @@ public function createCryptConfig(array $data)
     {
         $currentKey = $this->deploymentConfig->get(ConfigOptionsListConstants::CONFIG_PATH_CRYPT_KEY);
 
-        $configData = new ConfigData(ConfigFilePool::APP_ENV);
+        // TODO: refactor configData to factory in constructor
```

Comment:
I would suggest leaving the create as it is rather than moving it into the factory. You could always make a `getConfigData` method that checks for `$this->configData` being null and if so calls the factory create.

Reply:
Something like https://github.com/magento/magento2/pull/11152/files#diff-dfb3ee5a7f863458afea75f5524cd2e5R233 might work in this case also.